controllers/agentmachine_controller.go

```diff
@@ -236,6 +236,16 @@ func (r *AgentMachineReconciler) handleDeletionHook(ctx context.Context, log log
 		return nil, nil
 	}
 
+	// Skip agent unbind if AgentMachine is paused
```

Comment:
Are we sure we get here if things are paused?

IIRC (it's been a while) the idea here was to wait for CAPI to do the work it needs to do (i.e. drain/cordon the worker). At that point we will see the `clusterv1.PreTerminateDeleteHookSucceededCondition` condition set.

If something else CAPI-related is also watching this paused annotation will any of that happen?

Reply:
That's a good point. I think Hypershift is not setting any paused annotation on any pure CAPI resources such as `Cluster` or `Machine` but I'll double-check.

Reply:
The relevant code is in https://github.com/kubernetes-sigs/cluster-api/blob/be86b82e7e30a844bca141ff8bcdc450b0499549/internal/controllers/machine/machine_controller.go#L218

Reply:
I rechecked it and it turns out when we paused the `AgentCluster`, it actually does pause the CAPI generic `Cluster` CR and it hits this check and prevents the `Machine` from being deleted because reconciliation is paused.